scripts/eval/evaluate_rag.py

```
@@ -0,0 +1,1002 @@
+        # Check if more than 50% of queries failed
+        if self.error_count > total_questions * 0.5:
+            failure_file = os.path.join(self.result_dir, f"failure.txt")
+            with open(failure_file, 'w') as f:
+                f.write(f"Number of failures: {self.error_count} out of {total_questions} total queries\n")
+                f.write(f"Failure rate: {(self.error_count / total_questions) * 100:.2f}%\n")
+            print(f"⚠️ WARNING: High failure rate detected! {self.error_count} failures out of {total_questions} queries.")
+            print(f"Details written to {failure_file}")
+
+        return eval_data
+
+    def get_rag_response(self,query, collection_name, host, port, top_k, messages=[]):
```

Ruff (0.15.10)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)